### PR TITLE
update readme

### PR DESCRIPTION
**Pull request by jeremybumsted**
- updated plugin options
- updated examples

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -118,71 +135,79 @@ vault write data/buildkite/my_pipeline/git-credentials value=- <<< $(echo "https
...
+`method` (required)
+The auth method to use when authenticating with Vault. Currently only `approle` is supported
+
+`approle`
+Configures the plugin to use AppRole authentication to the Vault server. Requires a `role-id` be set.
+
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`role-id` (required)
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; The role-id the plugin should use to authenticate to Vault
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`secret-env`
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; The environment variable which holds the **secret-id** used to authenticate to vault. Default `VAULT_SECRET_ID`
```
Comment (toote) [inline README.md:165]: I don't think that this is a good way to show these options. Not only they are horrible in the source code, but also it kind of implies that the `approle` key should have a sub-dictionary with these keys.

File: README.md [hunk 2]
```diff
@@ -118,71 +135,79 @@ vault write data/buildkite/my_pipeline/git-credentials value=- <<< $(echo "https
...
 
-```bash
-vault write data/buildkite/my_pipeline/env_key value=- <<< $(echo "my secret"| base64)
-```
+Defaults to: `data/buildkite`
 
-Can be loaded using:
+
+### `auth`
+Configure the authentication parameters the plugin should use to authenticate with Vault.
```
Comment (toote) [inline README.md:152]: I would clarify that this is expected to be a dictionary/map and remove one of the two times `authenticate` is used for easier reading:
```suggestion
Dictionary/map with the configuration of the parameters the plugin should use to authenticate with Vault.
```

File: README.md [hunk 3]
```diff
@@ -118,71 +135,79 @@ vault write data/buildkite/my_pipeline/git-credentials value=- <<< $(echo "https
...
+
+`auth` accepts the following options:
```
Comment (toote) [inline README.md:154]: Technically, they are not options, just expected keys in the dictionary/map. As such, and to make them easier to reference I would add them with the same structure as other plugin options

File: README.md [hunk 4]
```diff
@@ -118,71 +135,79 @@ vault write data/buildkite/my_pipeline/git-credentials value=- <<< $(echo "https
...
+
+`approle`
+Configures the plugin to use AppRole authentication to the Vault server. Requires a `role-id` be set.
+
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`role-id` (required)
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; The role-id the plugin should use to authenticate to Vault
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`secret-env`
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; The environment variable which holds the **secret-id** used to authenticate to vault. Default `VAULT_SECRET_ID`
+
+
```
Comment (toote) [inline README.md:167]: As mentioned before, I would change the format of these keys to a similar one used for all other top-level plugin options:

```suggestion
#### `method` (required)

The auth method to use when authenticating with Vault. Currently only `approle` is supported

Possible values:
* `approle`: use AppRole authentication to the Vault server (requires a `role-id` be set)

#### `role-id` (required for `approle`)

The role-id the plugin should use to authenticate to Vault. Has no default value

#### `secret-env` (optional)

The environment variable which holds the **secret-id** used to authenticate to vault. Default `VAULT_SECRET_ID`
```

File: README.md [hunk 5]
```diff
@@ -118,71 +135,79 @@ vault write data/buildkite/my_pipeline/git-credentials value=- <<< $(echo "https
...
 ```bash
-.buildkite/steps/test_envvar.sh
+make integration-test
 ```
 
 When writing test plans, note that secrets are processed in the order they appear in the list returned from the Vault.
 
-## TODO
+### Testing the pipeline
+You can test the pipeline locally using the `bk cli`
```
Comment (toote) [inline README.md:206]: I would add a clarification here about why the additional environment variable is necessary

**Review by toote — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (jeremybumsted) [reply README.md]: Good call - removed 😅

File: README.md (hunk 5 above)
Comment (jeremybumsted) [reply README.md]: Good call - done!

**Review by toote — APPROVED** [review]

**Pull request merged**